internal/scheduler/scheduling_algo.go

```diff
@@ -190,42 +189,67 @@ func (l *LegacySchedulingAlgo) scheduleOnExecutor(
 		l.config,
 		totalCapacity,
 	)
-	queues := make([]*Queue, 0, len(priorityFactorByQueue))
-	for name, priorityFactor := range priorityFactorByQueue {
-		it := db.QueuedJobs(txn, name)
-		if queue, err := NewQueue(name, priorityFactor, &JobQueueIteratorAdapter{it: it}); err != nil {
-			return nil, err
-		} else {
-			queues = append(queues, queue)
-		}
-	}
-	legacyScheduler, err := NewLegacyScheduler(
-		ctx,
+	scheduler := NewRescheduler(
 		*constraints,
+		l.config,
+		&schedulerJobRepositoryAdapter{
+			txn: txn,
+			db:  db,
+		},
 		nodeDb,
-		queues,
-		totalResourceUsageByQueue,
+		priorityFactorByQueue,
+		allocatedByQueueAndPriority,
+		nil, // TODO: Add a repo to enable querying for scheduler reports.
 	)
+	result, err := scheduler.Schedule(ctx)
 	if err != nil {
 		return nil, err
 	}
-	result, err := legacyScheduler.Schedule(ctx)
-	if err != nil {
-		return nil, err
+
+	for i, job := range result.PreemptedJobs {
+		jobDbJob := job.(*jobdb.Job)
+		node := result.NodeByJobId[jobDbJob.GetId()]
+		if node == nil {
+			return nil, errors.Errorf("job %s not mapped to any node", jobDbJob.GetId())
+		}
+		if run := jobDbJob.LatestRun(); run != nil {
```

Comment:
I think in scheduler.go you error if a job is preempted but doesn't have a run. I think we should  be consistent here and either ignore in both case or error in both cases.

Reply:
Agreed. Fixed.